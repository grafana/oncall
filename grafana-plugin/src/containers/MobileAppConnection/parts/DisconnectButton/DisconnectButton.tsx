import React, { FC } from 'react';

import { Button } from '@grafana/ui';
import cn from 'classnames/bind';

import { WithConfirm } from 'components/WithConfirm/WithConfirm';

import styles from './DisconnectButton.module.scss';

const cx = cn.bind(styles);

type Props = {
  onClick: () => void;
};

export const DisconnectButton: FC<Props> = ({ onClick }) => (
  <WithConfirm title="Are you sure to disconnect your mobile application?" confirmText="Remove">
    <Button
      variant="destructive"
      onClick={onClick}
      size="md"
      className={cx('disconnect-button')}
      data-testid="test__disconnect"
    >
      Disconnect
    </Button>
  </WithConfirm>
);
