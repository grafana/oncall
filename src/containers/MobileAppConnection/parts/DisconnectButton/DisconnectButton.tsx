import React, { FC } from 'react';

import { Button, useStyles2 } from '@grafana/ui';
import { getUtilStyles } from 'styles/utils.styles';

import { WithConfirm } from 'components/WithConfirm/WithConfirm';

type Props = {
  onClick: () => void;
};

export const DisconnectButton: FC<Props> = ({ onClick }) => {
  const utilStyles = useStyles2(getUtilStyles);

  return (
    <WithConfirm title="Are you sure to disconnect your mobile application?" confirmText="Remove">
      <Button
        variant="destructive"
        onClick={onClick}
        size="md"
        className={utilStyles.centeredAbsolute}
        data-testid="test__disconnect"
      >
        Disconnect
      </Button>
    </WithConfirm>
  );
};
