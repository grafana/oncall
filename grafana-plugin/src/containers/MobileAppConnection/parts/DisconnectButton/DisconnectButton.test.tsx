import React from 'react';

import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { DisconnectButton } from './DisconnectButton';

describe('DisconnectButton', () => {
  test('it renders properly', () => {
    const component = render(<DisconnectButton onClick={() => {}} />);
    expect(component.container).toMatchSnapshot();
  });

  test('It calls the onClick handler when clicked', async () => {
    const mockedOnClick = jest.fn();

    render(<DisconnectButton onClick={mockedOnClick} />);

    // click the button, which opens the modal
    await userEvent.click(screen.getByRole('button'));
    // click the confirm button within the modal, which actually triggers the callback
    await userEvent.click(screen.getByText('Remove'));

    expect(mockedOnClick).toHaveBeenCalledWith();
    expect(mockedOnClick).toHaveBeenCalledTimes(1);
  });
});
