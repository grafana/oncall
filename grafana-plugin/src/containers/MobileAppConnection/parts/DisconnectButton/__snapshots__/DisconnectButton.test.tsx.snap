// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisconnectButton it renders properly 1`] = `
<div>
  <button
    class="css-ttl745-button disconnect-button"
    data-testid="test__disconnect"
    type="button"
  >
    <span
      class="css-1riaxdn"
    >
      Disconnect
    </span>
  </button>
</div>
`;
